src/token/TokenRejectTransaction.js

```diff
@@ -0,0 +1,270 @@
+ */
+
+/**
+ * @typedef {import("../channel/Channel.js").default} Channel
+ * @typedef {import("../client/Client.js").default<*, *>} Client
+ * @typedef {import("../transaction/TransactionId.js").default} TransactionId
+ */
+
+/**
+ * Reject a new Hedera™ crypto-currency token.
+ */
+export default class TokenRejectTransaction extends Transaction {
+    /**
+     *
+     * @param {object} [props]
+     * @param {AccountId | string | null} [props.owner]
+     * @param {HashgraphProto.proto.TokenReference[]} [props.rejections]
+     */
+    constructor(props = {}) {
+        super();
+
+        /**
+         * @private
+         * @type {?AccountId}
+         */
+        this._owner = null;
+
+        if (props.owner != null) {
+            this.setOwnerId(props.owner);
+        }
+
+        /**
+         * @private
+         * @type {TokenId[]}
+         */
+        this._tokenIds = [];
+
+        /**
+         * @private
+         * @type {NftId[]}
+         */
+        this._nftIds = [];
+
+        if (props.rejections != null) {
+            for (let rejection of props.rejections) {
+                if (rejection instanceof TokenId) {
+                    this._tokenIds.push(rejection);
+                } else if (rejection instanceof NftId) {
+                    this._nftIds.push(rejection);
+                }
+            }
+        }
+    }
+
+    /**
+     * @internal
+     * @param {HashgraphProto.proto.ITransaction[]} transactions
+     * @param {HashgraphProto.proto.ISignedTransaction[]} signedTransactions
+     * @param {TransactionId[]} transactionIds
+     * @param {AccountId[]} nodeIds
+     * @param {HashgraphProto.proto.ITransactionBody[]} bodies
+     * @returns {TokenRejectTransaction}
+     */
+    static _fromProtobuf(
+        transactions,
+        signedTransactions,
+        transactionIds,
+        nodeIds,
+        bodies,
+    ) {
+        const body = bodies[0];
+        const rejectToken =
+            /** @type {HashgraphProto.proto.ITokenRejectTransactionBody} */ (
+                body.tokenReject
+            );
+
+        return Transaction._fromProtobufTransactions(
+            new TokenRejectTransaction({
+                owner:
+                    rejectToken.owner != null
+                        ? AccountId._fromProtobuf(rejectToken.owner)
+                        : undefined,
+                rejections: rejectToken.rejections
+                    ? rejectToken.rejections
+                    : [],
+            }),
+            transactions,
+            signedTransactions,
+            transactionIds,
+            nodeIds,
+            bodies,
+        );
+    }
+
+    /**
+     * @returns {TokenId[] | string[]}
+     * @readonly
+     */
+    get tokenIds() {
+        return this._tokenIds;
+    }
+
+    /**
+     * @param {TokenId[] | string[]} tokenIds
+     * @returns {this}
+     */
+    setTokenIds(tokenIds) {
+        this._requireNotFrozen();
+        this._tokenIds = tokenIds.map((tokenId) =>
```

Comment:
Why do you need to handle `string[]` here?
I'd just go with `_tokenIds = tokenIds` or something like that.

Reply:
Because otherwise there would be two types in `this._tokenIds` might cause us some issue. So I'd like to keep it simple by only keeping values of type TokenId.

Reply:
This setter would be usually called(like in your integration tests) with passing a reference to some `tokenIds`. If you don't directly assign the passed `tokenIds` to `_tokenIds` and instead assign them one by one this function gets heavy for no reason in my opinion.